### ISSUE TITLE
Add a way to be warned when no translation is available

### ISSUE THREAD

**Issue by timkrief** [godotengine/godot-proposals#10430]
### Describe the project you are working on

I'm working on a fancy over engineered quiz game

### Describe the problem or limitation you are having in your project

I need to translate everything in the game, but it's hard to track what's missing.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It would be really helpful to enable a toggle to have a warning when users call `tr` or the built in objects tries to auto translate and fail, you would be notified if there is no translation for this message. It could also return the message it tried to translate.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

If `tr` is called internally each time a translation is needed, then it could be as simple as adding a warning printed to the console when no translation is available if you enabled a debug flag in the settings, in the `tr` function.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

To make this work I think I would have to recompile Godot itself to include the warning in the built in Object class function. So no, not really.

### Is there a reason why this should be core and not an add-on in the asset library?

It feels like a feature that any developer would need and I don't really know how to make that work without changing godot's code.

**Comment by Calinou** [comment]
This is already implemented in the form of [Pseudolocalization](https://docs.godotengine.org/en/stable/tutorials/i18n/pseudolocalization.html) 🙂

**Comment by timkrief** [comment]
> This is already implemented in the form of [Pseudolocalization](https://docs.godotengine.org/en/stable/tutorials/i18n/pseudolocalization.html) 🙂

I tried to use Pseudolocalization before opening this proposal, but I think it covers a different need. I'm not talking about strings that are not going through the `tr` function, I'm talking about string that do go through the `tr` function while running the game, and don't find any translation so they return an untranslated message. If I run my game in French for instance, some strings are still in English not because they are not going through the translation internally, but because there is no available translation for them. It's hard to spot them and a way to be warned in the console about "no translation found for message: *** " would be really handy.

**Comment by timkrief** [comment]
My current work around is to create a new translation with only "###" and to visually check what text is not "###" 
![image](https://github.com/user-attachments/assets/ad8a54bd-2010-4731-980c-42a81ea8c342)

**Comment by AdriaandeJongh** [comment]
Stating the obvious, but wouldn't there simply be empty strings in your translation source?

**Comment by timkrief** [comment]
> Stating the obvious, but wouldn't there simply be empty strings in your translation source?

Not if the translation is not even in the translation source. For instance if a button has a label "Continue" but "Continue" is not in the translation source at all in any language, it will just show up as "Continue" and the translation will be missing.

**Comment by AdriaandeJongh** [comment]
Right. If they go through node auto-translation or `tr()`, this _is_ precisely covered by setting up Pseudolocalization, making it instantly clear which labels aren't localized – it's just that those never produces any warnings. Correct?

**Comment by timkrief** [comment]
> Right. If they go through node auto-translation or `tr()`, this _is_ precisely covered by setting up Pseudolocalization, making it instantly clear which labels aren't localized – it's just that those never produces any warnings. Correct?

I answered this point here: https://github.com/godotengine/godot-proposals/issues/10430#issuecomment-2286329647



> I think it covers a different need. I'm not talking about strings that are not going through the `tr` function, I'm talking about string that do go through the `tr` function while running the game, and don't find any translation so they return an untranslated message.

**Comment by timkrief** [comment]
> here's a screenshot of the exact same functionality as your described workaround inside Pseudolocalization

It's not the same functionality as my workaround, my workaround would not replace strings that are not in my translation source, pseudolocalization would.
